dev-infrastructure/modules/metrics/actiongroups.bicep

```diff
@@ -3,22 +3,51 @@ import { csvToArray } from '../common.bicep'
 @description('Comma seperated list of email notifications. Only set in non MSFT environments!')
 param devAlertingEmails string
 
-@description('Comma seperated list of action groups for Sev 1 alerts.')
-param sev1ActionGroupIDs string
+@description('The ICM environment')
+param icmEnvironment string
 
-@description('Comma seperated list of action groups for Sev 2 alerts.')
-param sev2ActionGroupIDs string
+@description('Name of the ICM Action Group')
+param icmActionGroupName string
 
-@description('Comma seperated list of action groups for Sev 3 alerts.')
-param sev3ActionGroupIDs string
+@description('Name of the ICM Action Group')
+@maxLength(8)
+param icmActionGroupShortName string
 
-@description('Comma seperated list of action groups for Sev 4 alerts.')
-param sev4ActionGroupIDs string
+@description('ICM routing ID')
+param icmRoutingId string
 
-var sev1ActionGroups = csvToArray(sev1ActionGroupIDs)
-var sev2ActionGroups = csvToArray(sev2ActionGroupIDs)
-var sev3ActionGroups = csvToArray(sev3ActionGroupIDs)
-var sev4ActionGroups = csvToArray(sev4ActionGroupIDs)
+@description('ICM connection Name')
+param icmConnectionName string
+
+@description('ICM connection id')
+param icmConnectionId string
+
+@description('ICM automitigation enabled ID')
+param icmAutomitigationEnabled string
+
+resource icm 'Microsoft.Insights/actionGroups@2024-10-01-preview' = if (icmActionGroupName != '') {
+  name: 'icm-action-group'
+  location: 'global'
+  properties: {
+    enabled: true
+    groupShortName: icmActionGroupShortName
+    incidentReceivers: [
+      {
+        name: icmActionGroupName
+        incidentManagementService: 'Icm'
+        connection: {
+          name: icmConnectionName
+          id: icmConnectionId
+        }
+        mappings: {
+          'Icm.occurringlocation.environment': icmEnvironment
+          'Icm.routingid': icmRoutingId
+          'Icm.automitigationenabled': icmAutomitigationEnabled
+        }
+      }
+    ]
+  }
+}
 
 var emailAdresses = csvToArray(devAlertingEmails)
```

Comment:
Should we remove this or make it conditionally deployed.  Email based actiongroups are not really relevant to Microsoft environments.